File: shanny-blog-java/blog-pojo/src/main/java/com/shanny/entity/UserDetails.java
package com.shanny.entity;

import com.shanny.enums.UserEnum.*;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Data;
import org.springframework.format.annotation.DateTimeFormat;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "user_details")
public class UserDetails {
    @Id
    @Column(length = 50, nullable = false)
    @NotNull
    private String uuid;

    @Column(length = 30)
    private String nickname;

    @Column(length = 30)
    private String username;

    @DateTimeFormat(pattern = "yyyy-MM-dd")
    private LocalDate birthday;

    @Column(nullable = false)
    @NotNull
    @Enumerated(EnumType.STRING)
    private UserSex sex;

    @Column(length = 500)
    private String avatar;

    @Column(nullable = false)
    @NotNull
    @DateTimeFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime createTime;

    @Column(nullable = false)
    @NotNull
    @DateTimeFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime updateTime;
}

File: shanny-blog-net/blog-pojo/Entities/UserDetails.cs
using blog_common.Enums;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace blog_pojo.Entities
{
    [Table("user_details")]
    public class UserDetails
    {
        [Key]
        [MaxLength(50)]
        [Required]
        [Column("uuid")]
        public string Uuid { get; set; } = string.Empty;

        [MaxLength(30)]
        [Column("nickname")]
        public string Nickname { get; set; } = string.Empty;

        [MaxLength(30)]
        [Column("username")]
        public string Username { get; set; } = string.Empty;

        [Column("birthday")]
        public DateOnly? Birthday { get; set; }

        [Required]
        [Column("sex")]
        public UserSex Sex { get; set; }

        [MaxLength(500)]
        [Column("avatar")]
        public string Avatar { get; set; } = string.Empty;

        [Required]
        [Column("createTime")]
        public DateTime CreateTime { get; set; }

        [Required]
        [Column("updateTime")]
        public DateTime UpdateTime { get; set; }
    }
}
